pyaedt/generic/ibis_reader.py

```diff
@@ -0,0 +1,748 @@
+            if current_line == "":
+                break
+
+        ibis.components[component.name] = component
+
+    def fill_package_info(self, component, current_line, file):
+        """Extracts model's info.
+
+        Parameters
+        ----------
+        component : :class:`pyaedt.generic.ibis_reader.Component`
+            Current line content.
+        current_line : str
+            Current line content.
+        file : TextIO
+            File's stream.
+
+        """
+        while self.IsStartedWith(current_line, "|") == True or self.IsStartedWith(current_line, "[") == True:
+            current_line = file.readline()
+
+        if self.IsStartedWith(current_line, "R_pkg") == True:
+            component.R_pkg = current_line.strip()
+            current_line = file.readline()
+        elif self.IsStartedWith(current_line, "L_pkg") == True:
+            component.L_pkg = current_line.strip()
+            current_line = file.readline()
+        elif self.IsStartedWith(current_line, "C_pkg") == True:
+            component.C_pkg = current_line.strip()
+
+    def get_component_name(self, line):
+        """Gets the name of the component.
+
+        Parameters
+        ----------
+        line : str
+            Current line content.
+
+        Returns
+        -------
+        str
+            Name of the component.
+
+        """
+        name = ""
+        name = line.replace("[Component]", "")
+        return name.strip()
+
+    # Pin
+    def make_pin_object(self, line, component_name, ibis):
+        """Extracts model's info.
+
+        Parameters
+        ----------
+        line : str
+            Current line content.
+        component_name : str
+            Name of the component.
+        ibis : :class:`pyaedt.generic.ibis_reader.Ibis`
+            ibis object containing all info.
+
+        Returns
+        -------
+        :class:`pyaedt.generic.ibis_reader.Pin`
+            Pin object.
+
+        """
+
+        current_string = ""
+
+        current_string = line.strip().replace("\t", " ")
+        pin_name = self.get_first_parameter(current_string)
+        pin = Pin(pin_name + "_" + component_name + "_" + ibis.name, ibis.circuit)
+        pin.short_name = pin_name
+        current_string = current_string[len(pin.name) + 1 :].strip()
+        pin.signal = self.get_first_parameter(current_string)
+
+        current_string = current_string[len(pin.signal) + 1 :].strip()
+        pin.model = self.get_first_parameter(current_string)
+
+        current_string = current_string[len(pin.model) + 1 :].strip()
+        pin.r_value = self.get_first_parameter(current_string)
+
+        current_string = current_string[len(pin.r_value) + 1 :].strip()
+        pin.l_value = self.get_first_parameter(current_string)
+
+        current_string = current_string[len(pin.l_value) + 1 :].strip()
+        pin.c_value = self.get_first_parameter(current_string)
+
+        return pin
+
+    def get_first_parameter(self, line):
```

Comment:
This class method may not belong in this class as it does not depend on the state of the class. If you choose to have it in this class for style/organization, consider using `@classmethod`.